pkg/solana/chainwriter/transform_registry.go

```diff
@@ -89,13 +89,36 @@ func CCIPExecuteArgsTransform(ctx context.Context, client client.MultiClient, ar
 		return nil, nil, nil, fmt.Errorf("failed to fetch fee quoter address: %w", err)
 	}
 
+	if len(accounts) < MandatoryExecuteAccounts {
+		return nil, nil, nil, fmt.Errorf("encountered unexpected number of accounts, expected at least %d, got %d", MandatoryExecuteAccounts, len(accounts))
+	}
+
 	var tokenIndexes []uint8
 	// Accounts below are maintained to be in particular indexes in the Token Admin registry lookup table
 	poolProgram := poolLookupAccounts[2]
 	tokenProgram := poolLookupAccounts[6]
 	// Append token accounts to the account list and track at which index accounts for each token transfer starts
 	for _, message := range aggregatedMessages {
 		receiver := message.Receiver
```

Comment:
Blocking until this [discussion](https://chainlink-core.slack.com/archives/C0779R3AH8R/p1743214129717669?thread_ts=1742616847.916209&cid=C0779R3AH8R) resolves. If this is changed to the logic receiver, we need to extract the token receiver from the message to properly calculate `userTokenAccount` below.